### ISSUE TITLE
[Help Wanted] Rewrite build scripts in modern scripting language

### ISSUE THREAD

**Issue by 0x1mason** [0x1mason/GribApi.NET#85]
Build automation is mostly in _batch_, which, well... sucks. It's also not cross-platform, which is a barrier to supporting .NET Core/Mono. The build scripts should be re-written using one of the following, in rough order of preference:

* [_msbuild_](https://msdn.microsoft.com/en-us/library/0k6kkbsd.aspx)
* _bash_. Will need instructions in README for Windows users on how to add bash to PATH from _Git for Windows_, [_GnuWin32_ tools](https://chocolatey.org/packages/gnuwin32-coreutils.install), or other source. 
* pure [_python3_](https://www.python.org/downloads/) or [_Ninja_](https://pypi.python.org/pypi/ninja/1.7.2)
* NodeJS [_gulp_](https://gulpjs.com/)
* [_CSScript_](https://github.com/oleg-shilo/cs-script)
* Ruby [_Rake_](https://github.com/ruby/rake)
* [_gradle_](https://gradle.org/)
* [_PowerShell Core_](https://github.com/PowerShell/PowerShell)  (i.e., cross-platform)

If you want to use _make_, _Scala_, _groovy_, etc, I'm open to it. Anything other than _batch_.

**Comment by jnyrup** [comment]
I like [Cake](https://cakebuild.net/)
> Cake (C# Make) is a cross platform build automation system with a C# DSL to do things like compiling code, copy files/folders, running unit tests, compress files and build NuGet packages.

**Comment by HGCapacitor** [comment]
I did not look into the build system into detail, I agree that batch scripts is a bad idea ;)
Still what is the benefit in altering the build system as long as a NuGet package is produced. I do not think building in batch scripts interferes with targeting .net core (and making the NuGet package platform agnostic).
Moving away from the batch script will allow building the code on different platforms only.
And probably moving away from the batch scripts before upgrading to .net core will probably be less painfull.

In short:
-Allow building on multiple platforms (by moving away from the batch scripts) seems not a very interesting step to me at the moment
-Reduce the pain of configuring the build (by moving away from the batch scripts) is slightly more interesting, but I would currently put more weight in supporting BUFR files over upgrading to later .net frameworks
-I agree to putting msbuild on top of the prefered list
-As always please correct me if I am wrong!

**Comment by jnyrup** [comment]
@HGCapacitor For me the main point in moving to a dedicated build system is robustness and descriptive error messages, when some dependency is missing or fails to run.

**Comment by 0x1mason** [comment]
@HGCapacitor Agreed. I don't care that much as long as it is cross-platform, command-line friendly, and scriptable.

**Comment by 0x1mason** [comment]
This issue was moved to GribApiDotNet/GribApi.NET#8
